core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -245,9 +246,10 @@
     private final Highlighters highlighters = new Highlighters();
     private final Suggesters suggesters;
     /**
-     * Map from name to score parser.
+     * Map from name to score parser and it's ParseField.
      */
     private final Map<String, Tuple<ParseField, ScoreFunctionParser<?>>> scoreFunctionParsers = new HashMap<>();
+    private final ScoreFunctionsRegistry scoreFunctionsRegistry = new ScoreFunctionsRegistry(scoreFunctionParsers);
```

Comment:
I didn't think about this, maybe we should do the same with IndicesQueriesRegistry and get rid of the buildParserQueryRegistry method.